### DISCUSSION TITLE
issue: youtube error

### DISCUSSION THREAD

**Discussion by hongbo-miao** [open-webui/open-webui#7966] [Issues]
# Bug Report

## Installation Method

Helm Chart

## Environment

- **Open WebUI Version:** v0.4.1
- **Operating System:** Ubuntu 22.04

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

I am expecting YouTube video feature works when type `#` and add video URL, for example `#https://www.youtube.com/watch?v=PZRI1IfStY0`

## Actual Behavior:

> [ERROR: The content provided is empty. Please ensure that there is text or data present before proceeding.]

![image](https://github.com/user-attachments/assets/6b3ccf09-067c-4f66-814e-abb6b174c4e9)

## Description

**Bug Summary:**
[Provide a brief but clear summary of the bug]

## Reproduction Details

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

## Logs and Screenshots

**Browser Console Logs:**

![image](https://github.com/user-attachments/assets/2724e6cb-b900-4363-800f-b5ff6a578777)

**Docker Container Logs:**

```sh
INFO  [open_webui.apps.retrieval.main] save_docs_to_vector_db: document  74097b967a3bd94a36c87c54755423fd05706d0ed8bb17c94db3c620cb6bb54
ERROR [open_webui.apps.retrieval.main] The content provided is empty. Please ensure that there is text or data present before proceeding.
Traceback (most recent call last):
  File "/app/backend/open_webui/apps/retrieval/main.py", line 1071, in process_youtube_video
    save_docs_to_vector_db(docs, collection_name, overwrite=True)
  File "/app/backend/open_webui/apps/retrieval/main.py", line 786, in save_docs_to_vector_db
    raise ValueError(ERROR_MESSAGES.EMPTY_CONTENT)
ValueError: The content provided is empty. Please ensure that there is text or data present before proceeding.
INFO:     172.31.161.223:0 - "POST /retrieval/api/v1/process/youtube HTTP/1.1" 400 Bad Request
```

**Comment by sebaxakerhtc** [comment]
Same here
Docker installation
`POST /retrieval/api/v1/process/youtube HTTP/1.1" 400 Bad Request`

**Comment by tjbck** [comment]
Related: https://github.com/jdepoix/youtube-transcript-api/issues/303

![image](https://github.com/user-attachments/assets/8ca31ff8-6585-4645-a597-681b3e38a36e)

It most definitely works on my local machine, however I was able to confirm youtube import does not work my cloud instance.

**Comment by jakern** [comment]
I am getting the same error message using a local searxng web search. It _appears_ as though this error occurs in my case when it tries to fetch the same url for a second time. Are you able to confirm that you have the same issue on various youtube URLs?
